### ISSUE TITLE
IndexedDB persister

### ISSUE THREAD

**Issue by segments-tobias** [tinyplex/tinybase#69]
**Is your feature request related to a problem? Please describe.**
localStorage is limited to 5MB in size.

**Describe the solution you'd like**
[IndexedDB](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API) is a browser API for client-side storage of significant amounts of structured data. Since it's a database, I would think it could be a nice complement to TinyBase. Ideally, every change to TinyBase would be incrementally pushed to IndexedDB (maybe even in a Web Worker to unblock the main thread).

Do you know if anyone has explored this option? Are there any reasons this might not be a good idea?

**Comment by jamesgpearce** [comment]
Wow that is really awesome. I held back from doing this because I couldn't see a way to make it reactive to idb changes outside of TinyBase. Looks like you just stubbed that out, so this requires explicit saves and loads?

**Comment by yan-yanishevsky** [comment]
In my code above I made more code transformations so that in `idb` each row will be stored as a row of tinybase. It was easier to debug for me. 
If you don't need that, here is a much simpler example with `idb-keyval`. 

```ts
import { BroadcastChannel } from 'broadcast-channel'
import { get, set } from 'idb-keyval'
import { Store, createCustomPersister } from 'tinybase/cjs'

const BROADCAST_CHANNEL_NAME = 'tiny-tables-store'
const STORE_TABLES_NAME = 'tiny-tables-store'
const STORE_VALUES_NAME = 'tiny-values-store'

const broadcast = new BroadcastChannel(BROADCAST_CHANNEL_NAME)

export function createIdbPersister(store: Store) {
  const broadcastHandler = (event: string) => {
    if (event === 'reSync') {
      persister.load()
    }
  }
  const persister = createCustomPersister(
    store,
    async () => Promise.all([get(STORE_TABLES_NAME), get(STORE_VALUES_NAME)]),
    async (getContent) => {
      const [tables, values] = getContent()

      broadcast.postMessage('reSync')

      Promise.all([
        set(STORE_TABLES_NAME, tables),
        set(STORE_VALUES_NAME, values),
      ])
    },
    () => broadcast.addEventListener('message', broadcastHandler),
    () => broadcast.removeEventListener('message', broadcastHandler),
  )

  return persister
}
```

I also noticed just now, that I have a mistake there... 
`broadcast.postMessage('reSync')` should be in the setter function, not in the getter  (updated the comment)

**Comment by jamesgpearce** [comment]
Would you consider packaging this up as a pull request? Or are you OK if I implement this into the persister framework?

**Comment by yan-yanishevsky** [comment]
Yes, you can grab a version you like more

**Comment by jamesgpearce** [comment]
It's coming! https://github.com/tinyplex/tinybase/releases/tag/v4.2.0-beta.1 - please kick the tires and check it works.

Thank you so much @yan-yanishevsky for getting the ball rolling here 🙏

**Comment by jamesgpearce** [comment]
OK, released 4.2 with this in. Have fun and good luck!
